src/widgets/TagList.ts

```diff
@@ -0,0 +1,147 @@
+import { Spinner } from '@jupyterlab/apputils';
+import { Widget } from '@phosphor/widgets';
+import { AUTH_ERROR_MESSAGES } from '../git';
+import { Git, IGitExtension } from '../tokens';
+/**
+ * The UI for the content shown within the Git push/pull modal.
+ */
+export class GitTagDialog extends Widget {
```

Comment:
Could you please refractor the code, there are duplication and dead code in `GitTagDialog` and `GitTagCheckoutDialog`?

Moreover the AUTH_ERROR_MESSAGES has no reason to be used as the tag commands are all local commands that do not require authentication.